libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/NavigationMapRoute.java

```diff
@@ -601,50 +607,66 @@ public void removeRoute() {
 
   @Override
   public void onMapClick(@NonNull LatLng point) {
```

Comment:
IMO `onMapClick` is getting too long.
What about extracting blocks of code into well-named (based on the comments) `private` methods and simplifying conditional expressions?